File: 02-05-2025/cekPalindrom.go
package main

import "fmt"

func main1() {
	var input int
	fmt.Print("Input: ")
	fmt.Scan(&input)

	original := input
	reversed := 0

	for input > 0 {
		digit := input % 10
		reversed = reversed*10 + digit
		input = input / 10
	}

	if original == reversed {
		fmt.Printf("%d adalah palindrom\n", original)
	} else {
		fmt.Printf("%d bukan palindrom\n", original)
	}
}

File: Java/09_06_2025/cekPalindrom.java
import java.util.Scanner;

public class cekPalindrom {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.println("Program Cek Palindrom");

        System.out.print("Masukkan sebuah kata: ");
        String kata = input.nextLine();

        boolean isPalindrom = false;
        for (int i = 0; i < kata.length() / 2; i++) {
            if (kata.charAt(i) ==  kata.charAt(kata.length() - 1 - i)) {
                isPalindrom = true;
                break;
            }
        }

        if (isPalindrom) {
            System.out.println("Kata '" + kata + "' adalah palindrom.");
        } else {
            System.out.println("Kata '" + kata + "' bukan palindrom.");
        }

    }
}
